src/query/src/qExecutor.c

```diff
@@ -1541,7 +1541,7 @@ static void hashIntervalAgg(SOperatorInfo* pOperatorInfo, SResultRowInfo* pResul
   while (1) {
     int32_t prevEndPos = (forwardStep - 1) * step + startPos;
     startPos = getNextQualifiedWindow(pQueryAttr, &nextWin, &pSDataBlock->info, tsCols, binarySearchForKey, prevEndPos);
-    if (startPos < 0) {
+    if (startPos < 0 || startPos >= pSDataBlock->info.rows) {
```

Comment:
Break the loop if the `startPos` is no less than the number of retrieved rows ahead of time, which means that it had finished processing.

Reply:
This case might happen when `starPos = prevPosition + factor` for forwarding one step each time.